go/vt/vtctl/reparentutil/emergency_reparenter.go

```diff
@@ -199,7 +224,24 @@ func (erp *EmergencyReparenter) promoteNewPrimary(
 		return vterrors.Wrapf(primaryErr, "failed to PopulateReparentJournal on master: %v", primaryErr)
 	}
 
-	return nil
+	select {
+	case <-replSuccessCtx.Done():
+		// At least one replica was able to SetMaster successfully
+		return nil
+	case <-allReplicasDoneCtx.Done():
+		if len(rec.Errors) >= numReplicas {
```

Comment:
Why would summed up errors exceed the number of replicas? I think if it should never be greater than this is an odd thing to write in code.

Reply:
My thinking was that being more permissive here means the code still works if `handleReplica` is ever updated to record multiple errors per replica.

Reply:
I think if that's the case it should get updated when that happens. This code path implies this is a valid state of existence. (I'm required as a reader to read into the comment to understand that this is simply defensive programming, and for a reality we don't live in yet)

Reply:
or you could 
```
if len(rec.Errors) > numReplicas - 1
```

Reply:
I guess 
```
if len(rec.Errors) == numReplicas
```
feels weirdly specific, but I agree with @PrismaPhonic that if we ever record more than one error from each goroutine, this will need to change anyway. So it is probably best to check for the specific condition.

Reply:
> This code path implies this is a valid state of existence. 

The code directly states that this is an error state and not a valid state.

If I change this to `len(rec.Errors) == numReplicas` then we can return success in places where we definitely shouldn't.  Open to suggestions on how to make the comments more clear, though!

Reply:
How about:

```go
switch {
case len(rec.Errors) == numReplicas:
    // original error case in question
case len(rec.Errors) > numReplicas:
    return vterrors.Wrapf(rec.Error(), "received more errors (= %d) than replicas (= %d), which should be impossible", len(rec.Errors), numReplicas)
default:
    return nil
}
```

Reply:
This lgtm.

Reply:
I still am not seeing how we would return success in cases when we shouldn't if we truly think that errors will never exceeed count of num replicas. This does seem overly defensive to me, but I do think this newer version is better.